### PR TITLE
feat: update colors

### PR DESCRIPTION
**Pull request by nmerget**
## Types of changes

-   [ ] Bugfix (non-breaking change which fixes an issue)
-   [x] New feature (non-breaking change which adds functionality)
-   [ ] Refactoring (fix on existing components or architectural decisions)
-   [x] Breaking change (fix or feature that would cause existing functionality to not work as expected)
-   [ ] Documentation Update (if none of the other choices apply)

### CODE REVIEW

File: docs/migration/color-update.md
Comment (mfranzke) [inline docs/migration/color-update.md:1]: should't we call this `0.2.x-to-0.3.x.md` ?

Comment (mfranzke) [inline docs/migration/color-update.md:1]: @nmerget I'm loving that you've even already set up this document.

File: __snapshots__/card/patternhub/card-overview-should-match-screenshot.png
Comment (mfranzke) [inline __snapshots__/card/patternhub/card-overview-should-match-screenshot.png:1]: @nmerget is it expected that a new "functional" button would appear in this view?

File: __snapshots__/header/patternhub/header-properties-should-match-screenshot.png
Comment (mfranzke) [inline __snapshots__/header/patternhub/header-properties-should-match-screenshot.png:1]: @nmerget this doesn't look correct

File: __snapshots__/tag/component/chromium/DBTag-should-be-a-checkbox-checked.png
Comment (mfranzke) [inline __snapshots__/tag/component/chromium/DBTag-should-be-a-checkbox-checked.png:1]: the height of the tags seems to have changed – is this an expected change within this colors related PR ?

File: __snapshots__/tag/component/firefox/DBTag-should-be-a-checkbox.png
Comment (bruno-sch) [inline __snapshots__/tag/component/firefox/DBTag-should-be-a-checkbox.png:1]: To me, it looks like the new tag item is a touch higher at the top and bottom. Maybe one pixel on each side. Is that correct?

File: __snapshots__/tag/showcase/chromium/DBTag-should-match-screenshot-1/DBTag-should-match-screenshot.png
Comment (bruno-sch) [inline __snapshots__/tag/showcase/chromium/DBTag-should-match-screenshot-1/DBTag-should-match-screenshot.png:1]: Adapative and neutral are gray in the new version, previously they were white. What is correct?

File: __snapshots__/tooltip/showcase/chromium-highContrast/DBTooltip-should-match-screenshot-1/DBTooltip-should-match-screenshot.png
Comment (bruno-sch) [inline __snapshots__/tooltip/showcase/chromium-highContrast/DBTooltip-should-match-screenshot-1/DBTooltip-should-match-screenshot.png:1]: In the new screenshot, the expanded tooltip itself is not visible, only the button.

File: packages/components/src/components/tag/tag.scss [hunk 1]
```diff
@@ -73,6 +73,10 @@
 	&:has(db-button):has(dbbutton) .db-tab-remove-button {
 		display: none;
 	}
+
+	> [class^="db-"] {
+		block-size: 100%;
+	}
```
Comment (bruno-sch) [inline packages/components/src/components/tag/tag.scss:79]: What does this have to do with changing the colors?

File: packages/components/src/styles/_tag-components.scss [hunk 2]
```diff
@@ -7,27 +7,34 @@
...
 			&[data-emphasis="strong"] {
-				@extend %db-#{$name}-contrast-high;
+				--db-adaptive-on-bg-basic-emphasis-60-default: var(
+					--db-#{$name}-bg-inverted-contrast-high-default
+				);
+
+				@include colors.set-current-colors(
+					var(--db-#{$name}-bg-inverted-contrast-high-default),
+					var(--db-#{$name}-on-bg-inverted-default)
+				);
```
Comment (bruno-sch) [inline packages/components/src/styles/_tag-components.scss:37]: Can you explain what is happening here?

File: showcases/react-showcase/src/components/form/index.tsx
Comment (bruno-sch) [inline showcases/react-showcase/src/components/form/index.tsx:1]: What is happening here and what does this have to do with the change in colors?

File: __snapshots__/badge/showcase/chromium/DBBadge-should-match-screenshot-1/DBBadge-should-match-screenshot.png
[the remark this answers is not in the document]
Comment (nmerget) [reply __snapshots__/badge/showcase/chromium/DBBadge-should-match-screenshot-1/DBBadge-should-match-screenshot.png:1]: Yes the color changed since we first made the tag/badge component

File: packages/components/src/components/tag/tag.scss (hunk 1 above)
Comment (nmerget) [reply packages/components/src/components/tag/tag.scss:79]: Moved it: https://github.com/db-ui/mono/pull/2977

File: __snapshots__/tag/component/chromium/DBTag-should-be-a-checkbox-checked.png
Comment (nmerget) [reply __snapshots__/tag/component/chromium/DBTag-should-be-a-checkbox-checked.png:1]: moved it to: https://github.com/db-ui/mono/pull/2977

File: __snapshots__/card/patternhub/card-overview-should-match-screenshot.png
[the remark this answers is not in the document]
Comment (nmerget) [reply __snapshots__/card/patternhub/card-overview-should-match-screenshot.png:1]: It was an mistake, I removed it here, because it makes no sense

File: __snapshots__/tag/showcase/chromium/DBTag-should-match-screenshot-1/DBTag-should-match-screenshot.png
Comment (nmerget) [reply __snapshots__/tag/showcase/chromium/DBTag-should-match-screenshot-1/DBTag-should-match-screenshot.png:1]: The gray is correct, they should be bg-lvl-3 as default. Only if they are type `radio` or `checkbox` their initial state is bg-lvl-1

File: packages/components/src/styles/_tag-components.scss (hunk 2 above)
Comment (nmerget) [reply packages/components/src/styles/_tag-components.scss:37]: It's just a mixin to change `background-color` and `color` at the same time

File: docs/migration/color-update.md
Comment (nmerget) [reply docs/migration/color-update.md:1]: done

File: __snapshots__/header/patternhub/header-properties-should-match-screenshot.png
Comment (nmerget) [reply __snapshots__/header/patternhub/header-properties-should-match-screenshot.png:1]: fixed

File: __snapshots__/brand/patternhub/brand-properties-should-match-screenshot.png
[the remark this answers is not in the document]
Comment (nmerget) [reply __snapshots__/brand/patternhub/brand-properties-should-match-screenshot.png:1]: fixed

File: __snapshots__/header/patternhub/header-properties-should-match-screenshot.png
[the remark this answers is not in the document]
Comment (nmerget) [reply __snapshots__/header/patternhub/header-properties-should-match-screenshot.png:1]: fixed

File: __snapshots__/popover/showcase/chromium/DBPopover-should-match-screenshot-1/DBPopover-should-match-screenshot.png
[the remark this answers is not in the document]
Comment (nmerget) [reply __snapshots__/popover/showcase/chromium/DBPopover-should-match-screenshot-1/DBPopover-should-match-screenshot.png:1]: fixed

File: __snapshots__/tooltip/showcase/chromium-highContrast/DBTooltip-should-match-screenshot-1/DBTooltip-should-match-screenshot.png
Comment (nmerget) [reply __snapshots__/tooltip/showcase/chromium-highContrast/DBTooltip-should-match-screenshot-1/DBTooltip-should-match-screenshot.png:1]: fixed

File: __snapshots__/popover/showcase/chromium-highContrast/DBPopover-should-match-screenshot-1/DBPopover-should-match-screenshot.png
[the remark this answers is not in the document]
Comment (nmerget) [reply __snapshots__/popover/showcase/chromium-highContrast/DBPopover-should-match-screenshot-1/DBPopover-should-match-screenshot.png:1]: fixed

File: __snapshots__/tag/component/firefox/DBTag-should-be-a-checkbox.png
Comment (nmerget) [reply __snapshots__/tag/component/firefox/DBTag-should-be-a-checkbox.png:1]: fixed on main

File: packages/foundations/scss/colors/classes/_get-class.scss [hunk 3]
```diff
@@ -38,45 +18,37 @@
...
+			color: var(--db-adaptive-on-bg-basic-emphasis-90-default);
+		} @else if $color == "emphasis-80" {
+			color: var(--db-adaptive-on-bg-basic-emphasis-80-default);
+		} @else if $color == "emphasis-70" {
+			color: var(--db-adaptive-on-bg-basic-emphasis-70-default);
+		} @else if $color == "emphasis-60" {
+			color: var(--db-adaptive-on-bg-basic-emphasis-60-default);
+		} @else if $color == "emphasis-50" {
+			color: var(--db-adaptive-on-bg-basic-emphasis-50-default);
+		} @else {
```
Comment (mfranzke) [inline packages/foundations/scss/colors/classes/_get-class.scss:42]:
```suggestion
		$emphasis: emphasis-100, emphasis-90, emphasis-80, emphasis-70,
			emphasis-60, emphasis-50;
		@if list.index($emphasis, $color) != null {
			color: var(--db-adaptive-on-bg-basic-#{$color}-default);
		} @else {
```

@nmerget what do you think of this simplification? At the beginning of this file we would need an additional `@use "sass:list";`

File: packages/foundations/scss/colors/classes/_get-class.scss [hunk 4]
```diff
@@ -38,45 +18,37 @@
...
 			@error "Unknown color #{$color}.";
 		}
+		/* stylelint-enable at-rule-empty-line-before */
```
Comment (mfranzke) [inline packages/foundations/scss/colors/classes/_get-class.scss:45]:
```suggestion
```

would be obsolete if my previous code suggestion https://github.com/db-ui/mono/pull/2939/files#r1703937915 would get accepeted.

**Review by mfranzke — APPROVED** [review]

File: packages/foundations/scss/colors/classes/_get-class.scss (hunk 3 above)
Comment (nmerget) [reply packages/foundations/scss/colors/classes/_get-class.scss]: fixed

File: packages/foundations/scss/colors/classes/_get-class.scss (hunk 4 above)
Comment (nmerget) [reply packages/foundations/scss/colors/classes/_get-class.scss]: fixed

**Review by mfranzke — APPROVED** [review]
